src/encode/pattern/mod.rs

```diff
@@ -688,6 +699,21 @@ impl Deserialize for PatternEncoderDeserializer {
     }
 }
 
+/// Get the process ID as String for different OS
+#[cfg(not (target_os = "windows"))]
+fn get_pid() -> String { 
```

Comment:
Hmm, according to the doc it is a i32, see https://doc.rust-lang.org/regex/libc/type.pid_t.html I guess converting to a String is a safe bet at the Moment.